src/types/index.ts

```diff
@@ -58,3 +57,7 @@ export interface NavItem {
   href: string;
   label: string;
 }
+
+export type ActionState =
```

Comment:
Would move it to the `donateAction` file where it is used.

Also I think it is good to model the possible (legal) states separately in this case:
```ts
export type ActionState =
  | { status: "idle"; message?: string }
  | { status: "success"; message?: string }
  | { status: "error"; message?: string; issues?: Record<string, string[]> };
```